src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-list/rbd-list.component.ts

```diff
@@ -195,14 +195,32 @@ export class RbdListComponent extends ListWithDetails implements OnInit {
         this.getInvalidNameDisable(selection) ||
         selection.first().schedule_info === undefined
     };
+    const promoteAction: CdTableAction = {
+      permission: 'update',
+      icon: Icons.edit,
+      click: () => this.actionPrimary(true),
+      name: this.actionLabels.PROMOTE,
+      visible: () => (this.selection.first() != null && !this.selection.first().primary)
+        ? true : false
+    };
+    const demoteAction: CdTableAction = {
+      permission: 'update',
+      icon: Icons.edit,
+      click: () => this.actionPrimary(false),
+      name: this.actionLabels.DEMOTE,
+      visible: () => (this.selection.first() != null && this.selection.first().primary)
+       ? true : false
+    }
```

Comment:
Rather than creating a couple of actions for promoting/demoting (imperative approach), what about adding a checkbox `primary` in the RBD edit form (declarative)? This is a question open to everyone, @ceph/dashboard ?

In this very case I may see the point for each approach.